lib/dmv.rb

```diff
@@ -1,16 +1,21 @@
 class Dmv
+  attr_reader :facilities
 
   def initialize
     @facilities = []
   end
 
   def add_facility(facility)
-    @facilities << facility
+    @facilities.append(facility)
   end
 
   def facilities_offering_service(service)
-    @facilities.find do |facility|
-      facility.services.include?(service)
+    finalized = []
+    @facilities.each do |facility|
+      if facility.services.include?(service)
+        finalized.append(facility)
+      end
     end
+    finalized
```

Comment:
This does work, but all we had to do to fix the broken code was change the `find` to a `find_all` or a `select`. I encourage you to review the enumerables lesson to start using more enumerables other than .each.